### ISSUE TITLE
Missing *new* messages in old threads for new user with history turned off

### ISSUE THREAD

**Issue by igbjnI05bF** [vector-im/element-meta#2170]
### Steps to reproduce

This happens in any Element client with threads enabled, not just desktop app

1. Have a room with history turned off for new users
2. Have discussions in threads
3. New user joins the room, cannot see any history (as expected)
4. Have a different user send a *new* message in one of the old threads that existed before new user joined

#### What did you expect?
I expect to somehow see the message. I understand that I cannot see entire thread history, but I expect to see *new* messages. One way to handle this is for the first new message in the thread to create a new thread for the new user. The new user won't really know what the overall topic is about, but at least it's contained in a thread that the new user can participate in.
#### What happened instead?
I do not see any indication of the message at all. No notification, nothing. Disappears into the abyss. I *only* see the message if I disable threading in the client, then I see it in the normal timeline.

### Operating system

Linux OS and Android have been tested

### Homeserver

My new user account is on Conduit but I do not think this is a Conduit specific issue

### Will you send logs?

No

**Comment by t3chguy** [comment]
> My new user account is on Conduit but I do not think this is a Conduit specific issue

What makes you think it isn't? The client heavily relies on the server to provide threads as timelines via the /relations and /messages APIs. You unfortunately haven't even included logs so there's not much here to go on.

**Comment by igbjnI05bF** [comment]
I just tested with two Synapse accounts, same issue. Picture attached, both accounts Synapse. Image attached. Sorry for no logs but this is very easily reproducible:
1. Start a new room
2. Change room settings to only see history after join
3. Start a thread
4. Have new user join room
5. Have original user send a message in thread
6. Watch as absolutely no notification of anything happening happens

![elementerror](https://github.com/vector-im/element-web/assets/24235468/2c810f1e-d4da-4b84-a558-17697d57cc0f)

**Comment by andybalaam** [comment]
Chatted with @daniellekirkwood and this should follow the same pattern as https://github.com/vector-im/element-web/issues/26494

**Comment by daniellekirkwood** [comment]
Yep, when we worked on Threads first time around we discussed a similar thing to this [here](https://github.com/vector-im/element-meta/issues/1925). If a user doesn't have access to a root message or earlier messages (because it's not yet loaded, not scrolled back to, or restricted due to room events, etc.) then the error message should show: "Unable to load event that was replied to, it either does not exist or you do not have permission to view it."

(Ideally we'd have a different message but for now this is my MVP)
